semantic/solve.go

```diff
@@ -189,9 +190,14 @@ func unifyKindsByType(kinds map[Tvar]Kind, tv Tvar, t PolyType) (Substitution, e
 	}
 	switch k.(type) {
 	case ObjectKind, ArrayKind:
-		_, ok := t.(Tvar)
-		if !ok {
-			return nil, errors.New("invalid type for kind")
+		// TODO(jsternberg): This appears to be unused and to cause an error
+		// when the error is actually handled correctly.
+		// if _, ok := t.(Tvar); !ok {
+		// 	return nil, errors.New("invalid type for kind")
+		// }
+	case NullableKind:
```

Comment:
I guess there's no way to unify a nullable kind with a regexp type directly since kinds unify with kinds and types unify with types. I'm a little concerned that this might not catch all cases, since this method wasn't even being used.

Reply:
It might not. This function was used, but the error was never checked and the method has no side effects so I changed it to check for errors. It might be good to make it so this function actually returns a substitution and that substitution gets used so it is more than just a check.

Reply:
I don't think `unifyKindsByType` does anything useful and I think it should go away. In order to ensure that a regexp type is never nullable, I think we should define a `NonNullableKind`. Regexp types as well as composite types (functions, objects, etc...) should then be constrained to be non-nullable kinds. This would ensure that a regexp type and a nullable type would never pass unification.

Thoughts?